openedx/openedx-platform
# [SE-4304] feat: add celery beat configuration

This Pull Request adds `CELERYBEAT_SCHEDULER` to the edx-platform and sets it to its default value defined by Celery [documentation](https://docs.celeryproject.org/en/v4.4.7/userguide/configuration.html#beat-scheduler).

The configuration repository PR: https://github.com/edx/configuration/pull/6407

Settings

```yaml
ADDL_INSTALLED_APPS:
- django_celery_beat

EDXAPP_ENABLE_CELERY_BEAT: true
SANDBOX_ENABLE_REDIS: true
EDXAPP_CELERY_BROKER_TRANSPORT: 'redis'
EDXAPP_CELERY_BROKER_HOSTNAME: '127.0.0.1:6379'
EDXAPP_CELERY_BROKER_USE_SSL: false
EDXAPP_CELERY_BROKER_VHOST: '/0'
EDXAPP_CELERY_USER: 'default'
EDXAPP_CELERY_PASSWORD: ''
EDXAPP_CELERYBEAT_SCHEDULER: django_celery_beat.schedulers:DatabaseScheduler
```

**Dependencies**:

* https://github.com/edx/configuration/pull/6407

**Sandbox URL**: https://pr27630.opencraft.hosting/

**Merge deadline**: ASAP

**Testing instructions**:

* Login using the edX user
* Navigate to the Instructor dashboard's [download tab](https://pr27630.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-data_download)
* Click on Download profile information as a CSV
* Scroll down to the bottom of the page and wait some seconds
* Validate that the generated report appears and downloadable

**Author notes and concerns**:

TBD

**Reviewers**
